pkg/image/registry/imagestream/strategy.go

```diff
@@ -76,6 +79,16 @@ func (s Strategy) Validate(ctx kapi.Context, obj runtime.Object) field.ErrorList
 	}
 	errs := s.tagVerifier.Verify(nil, stream, user)
 	errs = append(errs, s.tagsChanged(nil, stream)...)
+
+	ns, ok := kapi.NamespaceFrom(ctx)
+	if !ok {
+		ns = stream.Namespace
+	}
+	err := s.limitVerifier.VerifyLimits(ns, stream)
+	if err != nil {
```

Comment:
Why? You're using the `err` variable only inside the if, having it written as:
```
if err := s.limitVerifier.VerifyLimits(ns, stream); err != nil {
    errs = append(errs, field.Forbidden(field.NewPath("imageStream"), err.Error()))
}
```
isn't any different, other than the fact that the `err` is scoped only to that if and not outside, which is what I meant with shadowing problems.